apps/condo/domains/notification/tasks/deliverMessage.js

```diff
@@ -46,130 +51,225 @@ async function _sendMessageByAdapter (transport, adapter, messageContext) {
+    const userId = get(user, 'id', null)
+    const userPhone = get(userData, 'phone', null)
+    const userEmail = get(userData, 'email', null)
+    const transports = new Set()
+
+    if (userId || remoteClient) transports.add(PUSH_TRANSPORT)
+    if (email || userEmail) transports.add(EMAIL_TRANSPORT)
+    if (phone || userPhone) transports.add(SMS_TRANSPORT)
+
+    return transports
 }
 
 /**
  * Tries to deliver message via available transports depending on transport priorities
  * based on provided message data and available channels. If more prioritized channels fail message delivery,
- * tries to delived message through less prioritized fallback channels. Updates message status & meta in every case.
+ * tries to deliver message through less prioritized fallback channels. Updates message status & meta in every case.
  * @param messageId
  * @returns {Promise<string>}
  */
 async function deliverMessage (messageId) {
-    const { keystone } = await getSchemaCtx('Message')
-    const message = await Message.getOne(keystone, { id: messageId })
+    const { keystone: context } = await getSchemaCtx('Message')
+    const message = await Message.getOne(context, { id: messageId })
 
-    if (message.id !== messageId) throw new Error('get message by id has wrong result')
+    if (isEmpty(message)) throw new Error('get message by id has wrong result')
     // Skip messages that are already have been processed
     if (!MESSAGE_SENDING_STATUSES[message.status]) return `already-${message.status}`
 
-    const baseAttrs = {
-        dv: message.dv,
-        sender: message.sender,
-    }
+    const baseAttrs = { dv: message.dv, sender: message.sender }
+    const { error } = await checkMessageTypeInBlackList(context, message)
 
-    const { error } = await checkMessageTypeInBlackList(keystone, message)
     if (error) {
-        return await Message.update(keystone, message.id, {
+        const messageErrorData = {
             ...baseAttrs,
             status: MESSAGE_ERROR_STATUS,
             processingMeta: {
                 dv: 1,
                 error,
             },
-        })
+        }
+
+        await Message.update(context, message.id, messageErrorData)
+
+        return MESSAGE_ERROR_STATUS
     }
 
-    const transports = await _choseMessageTransport(message)
-    const processingMeta = { dv: 1, transports, step: 'init' }
+    const processingMeta = { dv: 1 }
 
-    await Message.update(keystone, message.id, {
+    const messageInitData = {
         ...baseAttrs,
         status: MESSAGE_PROCESSING_STATUS,
         sentAt: null,
         deliveredAt: null,
         readAt: null,
         processingMeta,
-    })
+    }
+
+    await Message.update(context, message.id, messageInitData)
 
-    const transportsMeta = []
-    processingMeta.transportsMeta = transportsMeta
+    const userId = get(message, ['user', 'id'], null)
+    const options = _getMessageOptions(message)
+    const { strategy, defaultTransports } = options
```

Comment:
It’s too complex. Just get the transport from settings. If the settings is no set then use default. 
And move all this logic inside one function. You don’t need a special case for sms or email